tests/unit/test_control_plot_data.py

```diff
@@ -0,0 +1,120 @@
+import pytest
+import numpy as np
+from pandas import DataFrame
+from everviz.pages.controls import control_data_per_batch, control_data_initial_vs_best
+
+
+def test_control_data_per_batch(mocker):
+    api_mock = mocker.Mock()
+    control_values = [
+        {"control": "c1", "batch": 0, "value": 0},
+        {"control": "c2", "batch": 0, "value": 0},
+        {"control": "c1", "batch": 1, "value": 42},
+        {"control": "c1", "batch": 1, "value": 24},
+    ]
+    api_mock.control_values = control_values
+    result = control_data_per_batch(api_mock)
+    expected = DataFrame(control_values)
+
+    assert expected.equals(result)
+
+
+@pytest.mark.parametrize(
+    "controls, objectives, expected",
```

Comment:
Maybe use some module level constants?

Reply:
Do you mean like making a fixture out of them? so that it can be shared by multiple tests?

Reply:
Its just that the arguments of parametrize are so big, it gets harder to read. Putting them in a variable would make it more readable.